### DISCUSSION TITLE
Regex Path in Router

### DISCUSSION THREAD

**Discussion by vmtrandev** [uNetworking/uWebSockets.js#360] [Q&A] [answered]
Hi,
I am now migrating some code from express to uwebsocketjs. In express, I can define a path for specific router with this format
````javascript
router.get('/*.:postId', (req, res, next) => {
//some code here
})
````
I used regex to allow the request in format: /some-slug-string.postId (ex: /apple-iphone-8.9921785). The parameter should point to the part after [.] (postId) only. Unfortunately it doesn't seem to work in native uwebsocketjs router's path or I just did it wrong.I  came up with the alternative solution  which was to define a single "universal" router like ``` any('/*', (res, req) => {} )  ``` and add some logic code to  manually check the method, url,... but it will make the code way uglier.
I hope you can give me some suggestions.

**Comment by hst-m** [comment]
for the built-in router you should do:
```javascript
.get('/:phone/9921785', (res, req) => { const phone = req.getParameter(0) })
```
or
```javascript
.get('/:phone/:id', (res, req) => { const phone = req.getParameter(0), id = req.getParameter(1) })
```

**Comment by vmtrandev** [comment]
I guess you're misunderstanding something. ````apple-iphone-8.9921785```` is just an example for the pattern ````/*/````.  I mean it should accept any string but must contain the second part ````.9921785```` which is the unique id of the product so that I could make some queries to the DB. The usage of this concept is that I can rewrite the url to whatever I want, for example: ````apple-iphone-8-sale-20.9921785````, ````black-friday-apple-iphone-8.9921785````, and it will be more SEO-friendly.  Your given custom routing is what I've thought before, but imagine if I had 30+ http endpoints and I had to wrap all of them in switch-case block or if- else, I would be a true nightmare. I love the way uWebocketjs offers us a native routing so just wish to utilize it.

**Comment by ghost** [comment, marked as answer]
> I mean it should accept any string but must contain the second part .9921785

That would be /:whatever/9921785. There is no routing with "dot separator" only slash. So you either have to do it yourself in a get /* route or change from dot to slash.

**Comment by vmtrandev** [comment]
Alright thanks for your reply!
